src/modelz_llm/falcon_service.py

```diff
@@ -92,29 +116,177 @@ async def on_post(self, req: Request, resp: Response):
             resp.status = falcon.HTTP_400
             resp.data = ErrorResponse.from_validation_err(err, str(buf)).to_json()
             return
+        
+        temperature: float = 0.7
+        repetition_penalty: float = 1.2
+        top_p: float = 1.0
+        top_k: int = -1
+        logits_processor = prepare_logits_processor(
+            temperature, repetition_penalty, top_p, top_k
+        )
+
+        stop_token_ids = []
+        stop_token_ids.append(self.model.tokenizer.eos_token_id)
+
+        input_ids = self.model.tokenizer(chat_req.get_prompt(self.model_name)).input_ids
+        input_echo_len = len(input_ids)
+        output_ids = list(input_ids)
+        context_len = 2048
+        max_new_tokens = 512
+        is_encoder_decoder = True
```

Comment:
We should add it as a model attribute to tell the code if the model is a encoder-decoder model